notebooks/concepts/Mosviz concept notebook.ipynb

```diff
@@ -0,0 +1,542 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Alternative approach to loading with a \"master table\"\n",
+    "\n",
+    "If for whatever reason the above approach does not work, the data table itself can be viewed as a \"master dataset\".  Assuming the data objects are already loaded as above, that would look something like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "table = Table.read('/path/to/my/slit/catalog.ecsv')\n",
+    "\n",
+    "table['mosviz_spectra'] = spectra\n",
+    "table['mosviz_2d_spectra'] = spectra2d\n",
+    "table['mosviz_images'] = images\n",
+    "\n",
+    "mosapp.load_data(table)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Which would load the same state as the three method calls above.  If a user wanted the \"limited\" version (i.e., missing images or 2d spectra), they just don't add the relevant column to the table.\n",
+    "\n",
+    "This works because Astropy has a concept of \"mixin columns\" which let python objects with the appropriate leading dimension masquerade as table columns, so this might basically work as-is.  But it's not clear to me that all of the necessary data types support mixin columns right now, so this might require substantial underlying library work."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Custom loaders for \"other\" data "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We should support the option of using an I/O registry for custom loaders.  So then a user can write I/O functions like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from glob import glob\n",
+    "\n",
+    "@jdaviz.mosvizloader\n",
+    "def deimos_loader(mosvizapp, data_directory):\n",
+    "    spectrafn = glob(data_directory + _name'spec1d*.fits')\n",
+    "    SpectrumCollection([Spectrum1d.read(fn) for fn in spectrafn])\n",
+    "    \n",
+    "    spectra2dfns = glob(data_directory + _name'spec2d*.fits')\n",
+    "    spectra2d = [CCDData.read(fn, unit=u.adu) for fn in spectra2dfns]\n",
+    "    \n",
+    "    # there's no \"standard\" image format for deimos data, so we don't include images\n",
+    "    \n",
+    "    table = Table.read(data_directory + 'zspec.fits')\n",
+    "    \n",
+    "    mosvizapp.load_spectra(spectra)\n",
+    "    mosvizapp.load_2d_spectra(spectra2d)\n",
+    "    mosvizapp.load_table(table) "
```

Comment:
I don't understand how this would work - how would the data object know to tell the 2d viewer, etc that it should be updated? It seemed more natural to me to be implemented at some mosviz-helper level, because it depends on assuming the configuration has certain specific things.